kontrol-frontend/src/pages/TrafficConfiguration.tsx

```diff
@@ -12,40 +12,48 @@ const Page = () => {
   const prevResponse = useRef<string>();
   const { getTopology } = useApi();
   const { refetchFlows, flowVisibility } = useFlowsContext();
+  const timerRef = useRef<NodeJS.Timeout | null>(null);
 
-  useEffect(() => {
-    const fetchElems = async () => {
-      const response = await getTopology();
-      const filtered = {
-        ...response,
-        nodes: response.nodes.map((node) => {
-          return {
-            ...node,
-            versions: node.versions?.filter((version) => {
-              return flowVisibility[version.flowId] === true;
-            }),
-          };
-        }),
-      };
-      const newElems = utils.normalizeData(filtered);
-
-      // dont update react state if the API response is identical to the previous one
-      // This avoids unnecessary re-renders
-      if (JSON.stringify(newElems) === prevResponse.current) {
-        return;
-      }
-      prevResponse.current = JSON.stringify(newElems);
-      setElems(newElems);
-      // re-fetch flows if topology changes
-      refetchFlows();
+  const fetchElems = async () => {
+    const response = await getTopology();
+    const filtered = {
+      ...response,
+      nodes: response.nodes.map((node) => {
+        return {
+          ...node,
+          versions: node.versions?.filter((version) => {
+            return flowVisibility[version.flowId] === true;
+          }),
+        };
+      }),
     };
+    const newElems = utils.normalizeData(filtered);
+
+    // dont update react state if the API response is identical to the previous one
+    // This avoids unnecessary re-renders
+    if (JSON.stringify(newElems) === prevResponse.current) {
+      return;
+    }
+    prevResponse.current = JSON.stringify(newElems);
+    setElems(newElems);
+    // re-fetch flows if topology changes
+    refetchFlows();
+  };
+
+  const startPolling = () => {
+    timerRef.current = setInterval(fetchElems, pollingIntervalSeconds * 1000);
+  };
 
-    // Continuously fetch elements
-    const intervalId = setInterval(fetchElems, pollingIntervalSeconds * 1000);
-    fetchElems();
-    return () => clearInterval(intervalId);
+  const stopPolling = () => {
+    clearInterval(timerRef.current!);
+    timerRef.current = null;
+  };
```

Comment:
the TLDR of the change here is that the timer now uses a `useRef` instead of a `useEffect`, which has different behavior in regards to re-renders